### ISSUE TITLE
Element bounding box should be consistent across shapes

### ISSUE THREAD

**Issue by larvanitis** [evolus/pencil#127]
Currently there are multiple inconsistencies between the _real_ and _calculated_ bounding boxes (BB) of elements.
When I say _real_ I refer to the BB we get when the element is not part of a group. It is the BB that affects size and location.
On the other hand, when I say _calculated_ I am talking about the BB that affects the size and location when part of a group / multi-element selection.

This causes problems with resizing and moving grouped shapes, trying to match the size of different shapes etc. (see issue #100).

### Sketchy Shapes

As an example see the following screenshot of the really problematic case of sketchy shapes.

![bounding-box](https://cloud.githubusercontent.com/assets/1144686/17737198/8fab6d76-6494-11e6-854b-35996bbe6dc7.PNG)

Their BB should always be the same; IMHO the _real_ one, to be consistent with non-sketchy elements (see below).

---

### Different Shapes - Different BBs

The following shapes have:
**size:** 80x80 px
**background color:** orange
**border color:** yellow semi-transparent
**border with:** 10px

I have overlayed both the _real_ (blue) and the _calculated_ (green) BBs.
The red dotted lines are guides for comparison.

_Open image in new window for full size._

![bb-comparison-multiselect](https://cloud.githubusercontent.com/assets/1144686/17738786/73acec9c-649b-11e6-80ec-dd0cbe542298.PNG)

Assuming the CSS3 box-sizing property is used then we have different values for different shapes.

1. For example notice the smaller size of CS Rectangles which obviously account for the border while the rest of the shapes don't.

2. The BWE Pane has its border completely inside the shape while the rest of the shapes have it on top of their BB (half border in, half border out). Additionally notice the magical white space between the shape and its _calculated_ BB.

3. The _real_ vs _calculated_ BBs are all over the place. The prize in this category goes to CS Rectangle whose _calculated_ BB has different height and width.

---

### To Summarize

All shapes should have **consistent and predictable behavior**, preferably that of CS Oval. That is:
- The border should be around the shape's content (half in - half out).
- The border should not be included in the BBs.
- Both _real_ and _calculated_ BBs should match.

---

* Operating system: Linux
* Pencil version: 3.0.0-rc.1

**Comment by sTiKyt** [comment]
@larvanitis this repository seems outdated, please post your Issue to our fork! We'll updаte it soon!
https://github.com/DarkSecDevelopers/RePencil
